src/js/actions/layers.js

```diff
@@ -1528,6 +1529,37 @@ define(function (require, exports) {
     editVectorMask.writes = [locks.PS_DOC];
     editVectorMask.transfers = [tools.select];
 
+    var addVectorMask = function () {
+        var applicationStore = this.flux.store("application"),
+            currentDocument = applicationStore.getCurrentDocument(),
+            toolStore = this.flux.store("tool"),
+            superselectVector = toolStore.getToolByID("superselectVector");
+
+        if (currentDocument !== null) {
+            var layer = currentDocument.layers.selected.first(),
+            bounds = boundsLib.bounds(layer.bounds);
+
+            var payload = {
+                documentID: applicationStore.getCurrentDocumentID(),
+                layerIDs: Immutable.List.of(layer.id),
+                vectorMaskEnabled: true
+            };
+
+            this.dispatch(events.document.history.optimistic.ADD_VECTOR_MASK_TO_LAYER, payload);
+
+            return this.transfer(tools.select, superselectVector)
+                .bind(this)
+                .then(function () {
+                    return descriptor.batchPlayObjects([vectorMaskLib.makeBoundsWorkPath(bounds),
+                        vectorMaskLib.makeVectorMaskFromWorkPath(),
+                        vectorMaskLib.deleteWorkPath()]);
+                });
```

Comment:
They just added a vector mask, I assume here that they want to do something with it.